### PR TITLE
dt-shell-color-scripts: init at 1.0.0

### PR DESCRIPTION
**Pull request by jakobneufeld**
###### Description of changes
I am adding a new package, dt-shell-color-scripts. It is a bunch of color scripts that someone, dt, has collected. I followed the NixOS wiki tutorial on packaging shell scripts and I thought it would be a nice addition to nixpkgs.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by necrophcodr** [comment]
@jakobneufeld it looks like it doesn't entirely work yet? 

```
[steffen@necrophcodr-desktop:~/code/jakobneufield-nixpkgs]$ ./result/bin/colorscript -h
/run/current-system/sw/bin/find: ‘/opt/shell-color-scripts/colorscripts’: No such file or directory
/run/current-system/sw/bin/find: ‘/opt/shell-color-scripts/colorscripts’: No such file or directory
Description: A collection of terminal color scripts.

Usage: colorscript [OPTION] [SCRIPT NAME/INDEX]
  -h, --help, help      Print this help.                                      
  -l, --list, list      List all installed color scripts.                     
  -r, --random, random  Run a random color script.                            
  -e, --exec, exec      Run a specified color script by SCRIPT NAME or INDEX. 
  -b, --blacklist, blacklist    Blacklist a color script by SCRIPT NAME or INDEX.     
  -u, --unblacklist, unblacklist        Unblacklist a color script by SCRIPT NAME or INDEX.   
  -a, --all, all        List the outputs of all colorscripts with their SCRIPT NAME
```

Looks like there's something not quite working with the path to the colorscripts directory. You may want to perform a `substituteInPlace ./colorscript.sh "/opt/shell-color-scripts/colorscripts" "$out/opt/shell-color-scripts/colorscripts"` perhaps, although I have not tested that.

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}:
+  stdenv.mkDerivation {
+    pname = "dt-shell-color-scripts";
+    version = "1.0.0";
+    src = fetchFromGitLab {
+      # https://gitlab.com/dwt1/shell-color-scripts
+      owner = "dwt1";
+      repo = "shell-color-scripts";
+      sha256 = "sha256-bd3NBf99rCiADUKQb6fzCBDaKBmYaZHcO4qokm/39do=";
+      # have to use revesion because this package has no version tags so I am pinning the latest commit
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:15]: There is no need to apologize here. There are many packages on Nixpkgs that use Git hashes.

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+, makeWrapper
+}:
+  stdenv.mkDerivation {
+    pname = "dt-shell-color-scripts";
+    version = "1.0.0";
+    src = fetchFromGitLab {
+      # https://gitlab.com/dwt1/shell-color-scripts
+      owner = "dwt1";
+      repo = "shell-color-scripts";
+      sha256 = "sha256-bd3NBf99rCiADUKQb6fzCBDaKBmYaZHcO4qokm/39do=";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:14]: Content hashes should be the last attribute.

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+, lib
+, fetchFromGitLab
+, bash
+, makeWrapper
+}:
+  stdenv.mkDerivation {
+    pname = "dt-shell-color-scripts";
+    version = "1.0.0";
+    src = fetchFromGitLab {
+      # https://gitlab.com/dwt1/shell-color-scripts
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:11]: What the hell is this comment-URL doing here?

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      owner = "dwt1";
+      repo = "shell-color-scripts";
+      sha256 = "sha256-bd3NBf99rCiADUKQb6fzCBDaKBmYaZHcO4qokm/39do=";
+      # have to use revesion because this package has no version tags so I am pinning the latest commit
+      rev = "fcd013ea2e1ff80e01adbcea9d0eaf9c73db94c0";
+    };
+    meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:18]: meta should be the last attribute on derivation.

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      description = "A collection of shell color scripts collected by dt (Derek Taylor)";
+      homepage = "https://gitlab.com/dwt1/shell-color-scripts";
+      license = with licenses; [ mit ];
+      maintainers = with maintainers; [ drzoidberg ];
+    };
+    buildInputs = [ bash ];
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:24]: Don't be afraid of using spaces.

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    nativeBuildInputs = [ makeWrapper ];
+    installPhase = ''
+      # Adjust shebangs in all files to comform to nix stadards
+      find . -type f -exec sed -i "1s/.*/#!\/usr\/bin\/env bash/" {} +
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:28]: This is the most unclean `patchShebangs` I have ever seen.

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, bash
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:4]: Use `runtimeShell` instead of Bash.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix (hunk 6 above)
Comment (jakobneufeld) [reply pkgs/tools/misc/dt-shell-color-scripts/default.nix:28]: did not know that patchShebangs existed

**Comment by jakobneufeld** [comment]
For some wierd reason
my commit history got messed up

**Comment by jakobneufeld** [comment]
I will make a new pull request without all these wierd quirks. I am terribly sorry.

**Pull request closed without merging**